src/utils.js

```diff
@@ -29,3 +29,29 @@ export function createHorizontalLine () {
 	hLine.appendChild(document.createTextNode('.'))
 	return hLine
 }
+
+export function recalculateActive (terminalsSet, active) {
+	const terminals = [...terminalsSet]
+	terminals.sort((a, b) => {
+		// active before other
+		if (active && a === active) {
+			return -1
+		}
+		if (active && b === active) {
+			return 1
+		}
+		// visible before hidden
+		if (a.isVisible() && !b.isVisible()) {
+			return -1
+		}
+		if (!a.isVisible() && b.isVisible()) {
+			return 1
+		}
```

Comment:
Cant this be 

```js
export function recalculateActive (terminalsSet, active) {
	const terminals = [...terminalsSet]
	terminals.sort((a, b) => {
		// active before other || visible before hidden
		if (active && a === active) || a.isVisible() && !b.isVisible() {
			return -1
		}
		if (active && b === active) || !a.isVisible() && b.isVisible() {
			return 1
		}
```

Reply:
ya but I thought it was easier to read with the conditions spread out.

Reply:
Its probably nothing ins the evaluation time, I thought it was also easier to read as mentioned.

As a JS newb I cant say I find it hard to read either way, just seems more efficient not to dupe unnecessary returns

Reply:
The JS engine will most likely optimize the code the same way no matter which way we write it. I heard someone say one time "If you think you are smarter than the compiler/interpreter, you're most likely wrong."

Reply:
Im sure there not much in it like I said `ts probably nothing added/lost in the evaluation time either way.` that means its probably the same no matter how its written

The reason you gave was readability and I dont think it actually improves the readability.

Reply:
And the logic is changed in your code. If `b === active && a.isVisible() && !b.isVisible()` my code will return `1` like it should and your code will return `-1`. The active terminal should come before any other even if it is not visible.